src/index.d.ts

```diff
@@ -0,0 +1,177 @@
+import { BufferGeometry, EdgesGeometry, Group, InstancedMesh, LineSegments, Mesh, MeshPhongMaterial, Plane, Triangle, Vector3 } from 'three';
+
+export class Brush extends Mesh {
+
+  isBrush: boolean;
+  markUpdated(): void;
+  isDirty(): boolean;
+  prepareGeometry(): void;
+  disposeCacheData(): void;
+
+}
+
+export class TypedAttributeData {
+
+  groupAttributes: Object[];
+  groupCount: number;
+
+  getType( name: String ): string;
+  getTotalLength( name: String ): number;
+  getGroupSet( index?: number ): Object;
+  getGroupArray( name: string, index?: number ): Array; // return Array ???
+  initializeArray( name: string, type: string ): void;
+  clear(): void;
+  delete( key: string ): void;
+  reset(): void;
+
+}
+
+export enum OperationStrategy{} // Is the name correct?
+export const ADDITION: OperationStrategy;
+export const SUBTRACTION: OperationStrategy;
+export const DIFFERENCE: OperationStrategy;
+export const INTERSECTION: OperationStrategy;
+
+export class Evaluator {
+
+  triangleSplitter: TriangleSplitter;
+  attributeData: TypedAttributeData;
+  attributes: String[];
+  useGroups: boolean;
+  debug: OperationDebugData;
+
+  evaluate( a: Brush, b: Brush, operation: OperationStrategy, targetBrush?: Brush ): Brush;
+
+  evaluateHierarchy( root: Brush, target?: Brush ): Brush; // Root is Brush type??
```

Comment:
"root" should be "Operation" or "OperationGroup" here - heh this is good you're getting to things I haven't managed to document, yet 😁

Reply:
Ups, sorry 🤭!! I tried to understand the code and guess which are the correct types. 
If you think it is better I remove them.

Reply:
No it's great! Thanks for including so much!